src/V3Width.cpp

```diff
@@ -6520,6 +6520,61 @@ class WidthVisitor final : public VNVisitor {
         }
     }
 
+    bool areAggregatesComparable(AstNodeDType* lhs, AstNodeDType* rhs) {
+        if (!lhs || !rhs) return false;
+
+        lhs = lhs->skipRefp();
+        rhs = rhs->skipRefp();
+
+        if (VN_IS(lhs, UnpackArrayDType) && VN_IS(rhs, UnpackArrayDType)) {
+            auto* lhsp = VN_CAST(lhs, UnpackArrayDType);
+            auto* rhsp = VN_CAST(rhs, UnpackArrayDType);
+
+            // Compare element types
+            if (!areAggregatesComparable(lhsp->subDTypep(), rhsp->subDTypep())) return false;
+
+            // Compare size
+            std::vector<AstUnpackArrayDType*> lhsDims = lhsp->unpackDimensions();
+            std::vector<AstUnpackArrayDType*> rhsDims = rhsp->unpackDimensions();
+            if (lhsDims.size() != rhsDims.size()) return false;
+            for (size_t i = 0; i < lhsDims.size(); ++i) {
+                int lsz = lhsDims[i]->elementsConst();
+                int rsz = rhsDims[i]->elementsConst();
+                if (lsz >= 0 && rsz >= 0 && lsz != rsz) return false;
+            }
+
+            return true;
+        }
+
+        if (VN_IS(lhs, DynArrayDType) && VN_IS(rhs, DynArrayDType)) {
+            return areAggregatesComparable(lhs->subDTypep(), rhs->subDTypep());
+        }
+
+        if (VN_IS(lhs, QueueDType) && VN_IS(rhs, QueueDType)) {
+            return areAggregatesComparable(lhs->subDTypep(), rhs->subDTypep());
+        }
+
+        if (VN_IS(lhs, AssocArrayDType) && VN_IS(rhs, AssocArrayDType)) {
+            const auto* lhsp = VN_CAST(lhs, AssocArrayDType);
+            const auto* rhsp = VN_CAST(rhs, AssocArrayDType);
+
+            if (!areAggregatesComparable(lhsp->subDTypep(), rhsp->subDTypep())) return false;
+
+            if (!lhsp->keyDTypep()->sameTree(rhsp->keyDTypep())) return false;
```

Comment:
```suggestion
            if (!lhsp->keyDTypep()->similarDTypes(rhsp->keyDTypep())) return false;
```

Otherwise typedefs will not be equal.  Please add some checks that use two different typedefs pointing to same thing in your test cases.